docusaurus/docs/introduction/plugin-lifecycle.md

```diff
@@ -0,0 +1,93 @@
+---
+id: plugin-lifecycle
+title: Life cycle of a plugin
+description: Learn about the life cycle of a Grafana plugin.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - lifecycle
+  - life cycle
+  - loading
+  - unloading
+  - installation
+sidebar_position: 2.5
+---
+
+# Life cycle of a plugin
+
+This document describes the various phases of a plugin including installation and loading. We will describe the differences in a plugin's life cycle depending on its type and whether or not it has a backend.  
+
+:::note
+
+The life cycle of plugins is tracked in-memory and is not persisted in Grafana’s database. 
+
+:::
+
+## Installing and uninstalling a plugin
+
+For instructions to install or uninstall plugins, see our documentation at [Plugin administration](https://grafana.com/docs/grafana/latest/administration/plugin-management/#install-grafana-plugins).
+
+Upon installation, a plugin is extracted to the _plugin directory_ on the filesystem. Similarly, uninstalling a plugin deletes the files from the same directory.  
+
+:::note
+
+Use of the Grafana CLI to install or uninstall a plugin requires you to restart Grafana for the change to take effect. To avoid restarting Grafana, you can install a plugin from within the Grafana [plugin catalog](https://grafana.com/plugins/) during runtime.
+
+:::
+
+## Loading plugins
+
+Understanding the different phases involved when Grafana is starting a plugin may help you better understand plugin usage and to debug any unexpected behavior during loading.
```

Comment:
Is there anything more we could offer for users to `debug any unexpected behavior during loading`

What are the errors they may see, where will they see them?

Reply:
@marefr - Any thoughts on this? I would like to see more detail, or perhaps we remove?

Reply:
I started to look at what information is logged when Grafana starts up and noticed that we lacked some standard info log messages why I opened https://github.com/grafana/grafana/pull/76896. 

I think it make sense to keep this, but I would suggest referring to inspecting the Grafana server log somehow with optional debug level enabled. 

> What are the errors they may see, where will they see them?

I think it might be hard to explain in detail any errors that could happen.

Something like below perhaps. WDYT?

```suggestion
Plugins are loaded either when Grafana starts up or when a plugin has been installed/uninstalled during runtime. 

Understanding the different phases involved when Grafana is loading a plugin may help you better understand plugin usage and troubleshoot any unexpected behavior, such as why a certain plugin is not marked as installed in the plugins catalog and/or for use within Grafana even though you've installed it. You can check the [Grafana server log](https://grafana.com/docs/grafana/latest/troubleshooting/#troubleshoot-with-logs) for any unexpected errors or details related to loading a plugin. In addition, you can enable even more details by changing the [log level to debug](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#log).
```